source/aggregates-bdd-api.coffee

```diff
@@ -22,16 +22,35 @@ class AggregateTest
       return data
 
   given: (data) ->
-    if _.isArray(data)
-      # We have to add a commit with the historic events
-      changes = events: data, commands: []
-      aggregateId = data[0].sourceId
-      version = data[0].version ? 1
-      @_commitStore.add changes, aggregateId, version - 1
-    else if data instanceof Space.messaging.Command
-      # We just send the command through the app and let
+    if data is undefined then return this
+    commands = []
+    events = []
+    if !_.isArray(data)
```

Comment:
Maybe you could just wrap the data into an array like `data = [data]` when it's not an array :wink: so you just need the array handling code to setup the commands / events.